src/03_primitives.jl

```diff
@@ -262,5 +262,6 @@ end
 
 # ----- Base.reverse ---------------------------------------------------------#
 
-# Use ready-made method for AbstractArrays or implement method specific to
-# NullableArrays, possibly for performance purposes?
+function Base.reverse(X::NullableVector, s=1, n=length(X))
+    return(NullableArray(reverse(X.values, s, n), reverse(X.isnull, s, n)))
```

Comment:
This method is potentially unsafe if the eltype of X is not isbits. How does Base handle undefined entries in the array being reversed?

Also, should we provide a mutating version and then define the pure version in terms of the mutating version?